### ISSUE TITLE
Unexpected warnings for nullable value types

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#31904]
**Version Used**: Microsoft.Net.Compilers 3.0.0-beta2-18616-01

**Steps to Reproduce**:

```csharp
private static int? _seed = null;

public static int? Seed
{
    get
    {
        if (_seed.HasValue)
        {
            return _seed.Value; // CS8629??
        }
        else
        {
            return null;
        }
    }

    set
    {
        if (!_seed.HasValue)
        {
            _seed = value;
            if (_seed.HasValue)
            {
                Debug.WriteLine("Seeding Random with: " + _seed.Value.ToString()); // CS8629??
                _rand = new Random(_seed.Value); // CS8629??
            }
        }
        else
        {
            Debug.WriteLine("Attempt to seed Random to " + value.ToString() + " rejected it was already seeded to: " + _seed.Value.ToString()); // CS8629??
        }
    }
}
```

**Expected Behavior**:

CS8629 is not reported.

**Actual Behavior**:

CS8629 is reported at the commented lines.

📝 The warnings are still reported if the `HasValue` expressions are replaced with the equivalent null-check syntax.

**Comment by cston** [comment]
The underlying issue is we're not tracking static members currently. See https://github.com/dotnet/roslyn/issues/26651.

**Comment by sharwell** [comment]
Duplicate of #26651
